### PR TITLE
[Example] Add ARMA for Node Classification

### PR DESCRIPTION
**Pull request by xnuohz**
## Description
Pytorch Implementation of ARMA in paper [Graph Neural Networks with convolutional ARMA filters](https://arxiv.org/abs/1901.01343). @mufeili 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/pytorch/arma/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# DGL Implementation of ARMA
+
+This DGL example implements the GNN model proposed in the paper [Graph Neural Networks with convolutional ARMA filters](https://arxiv.org/abs/1901.01343). For the original implementation, see [here](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.ARMAConv).
```
Comment (mufeili) [inline examples/pytorch/arma/README.md:3]: "For the original implementation, see [here](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.ARMAConv)."

Is this the implementation from the authors? If not, there's no need to mention it.

File: examples/pytorch/arma/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+Contributor: [xnuohz](https://github.com/xnuohz)
+
+### Requirements
+The codebase is implemented in Python 3.6. For version requirement of packages, see below.
+
+```
+dgl 0.6a210202
```
Comment (mufeili) [inline examples/pytorch/arma/README.md:11]: "0.6a210202" is not a formal version. Either you specify a formal version or just not mention it.

File: examples/pytorch/arma/README.md (hunk 1 above)
Comment (xnuohz) [reply examples/pytorch/arma/README.md:3]: They mentioned PyG public implementation before section 6.1, but this is not from the authors.

File: examples/pytorch/arma/README.md (hunk 2 above)
Comment (xnuohz) [reply examples/pytorch/arma/README.md:11]: Fine, the version will be removed.

File: examples/pytorch/arma/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+--hid-dim          int     Hidden layer dimensionalities.      Default is 16.
+--num-stacks       int     Number of K.                        Default is 2.
+--num-layers       int     Number of T.                        Default is 1.
+--dropout          float   Dropout applied at all layers.      Default is 0.75.
+```
+
+###### Examples
+
+The following commands learn a neural network and predict on the test set.
+Train a ARMA model which follows the original hyperparameters on different datasets.
```
Comment (mufeili) [inline examples/pytorch/arma/README.md:58]: "a ARMA" -> "an ARMA"

File: examples/pytorch/arma/README.md (hunk 1 above)
Comment (mufeili) [reply examples/pytorch/arma/README.md]: I think you don't need to mention it as it's not from the authors.

File: examples/pytorch/arma/citation.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    acc_lists = []
+
+    for _ in range(50):
+        acc_lists.append(main(args))
+
+    acc_lists = list(reversed(sorted(acc_lists)))
+    acc_lists_top = np.array(acc_lists[:10])
+
+    mean = np.around(np.mean(acc_lists_top, axis=0), decimals=3)
+    std = np.around(np.std(acc_lists_top, axis=0), decimals=3)
```
Comment (mufeili) [inline examples/pytorch/arma/citation.py:148]: Did the authors do so?

File: examples/pytorch/arma/citation.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
+""" The main file to train a ARMA model using a full graph """
```
Comment (mufeili) [inline examples/pytorch/arma/citation.py:1]: "a ARMA" -> "an ARMA"

File: examples/pytorch/arma/citation.py (hunk 4 above)
Comment (xnuohz) [reply examples/pytorch/arma/citation.py:148]: The paper didn't mention the detail; PyG version code run 100 times.

File: examples/pytorch/arma/model.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import dgl
```
Comment (mufeili) [inline examples/pytorch/arma/model.py:4]: Not used

**Comment by mufeili** [comment]
Done a first pass.

File: examples/pytorch/arma/citation.py (hunk 4 above)
Comment (mufeili) [reply examples/pytorch/arma/citation.py:148]: Did they pick the top-10 accuracies?

Comment (xnuohz) [reply examples/pytorch/arma/citation.py:148]: They used all results.

**Review by mufeili — APPROVED** [review]
LGTM
